modules/SofaGraphComponent/src/SofaGraphComponent/SceneCheckCollisionResponse.cpp

```diff
@@ -0,0 +1,113 @@
+/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#include "SceneCheckCollisionResponse.h"
+
+#include <sofa/simulation/Node.h>
+#include <SofaBaseCollision/DefaultContactManager.h>
+#include <sofa/core/behavior/BaseAnimationLoop.h>
+#include <sofa/core/behavior/ConstraintSolver.h>
+
+namespace sofa::simulation::_scenechecking_
+{
+
+using sofa::simulation::Node;
+
+const std::string SceneCheckCollisionResponse::getName()
+{
+    return "SceneCheckCollisionResponse";
+}
+
+const std::string SceneCheckCollisionResponse::getDesc()
+{
+    return "Check there is not duplicated name in the scenegraph";
+}
+
+void SceneCheckCollisionResponse::doInit(Node* node)
+{
+    SOFA_UNUSED(node);
+    m_message.str("");
+}
+
+void SceneCheckCollisionResponse::doCheckOn(Node* node)
+{
+    if(m_checkDone)
+        return;
+
+    const sofa::core::objectmodel::BaseContext* root = node->getContext()->getRootContext();
+    std::vector<sofa::component::collision::DefaultContactManager*> contactManager;
+    root->get<sofa::component::collision::DefaultContactManager>(&contactManager, core::objectmodel::BaseContext::SearchDown);
+    Size nbContactManager = contactManager.size();
+    if( nbContactManager == 0 )
+    {
+        m_checkDone=true;
+    }
+    else
+    {
+        m_checkDone=true;
```

Comment:
```suggestion
    m_checkDone=true;
    const Size nbContactManager = contactManager.size();
    if( nbContactManager  > 0 )
    {
```
`m_checkDone` is set to `true` in any case